include/libtorrent/announce_entry.hpp

```diff
@@ -62,13 +91,7 @@ namespace libtorrent
 		// trackerid is sent).
 		std::string trackerid;
 
-		// if this tracker has returned an error or warning message
-		// that message is stored here
-		std::string message;
-
-		// if this tracker failed the last time it was contacted
-		// this error code specifies what error occurred
-		error_code last_error;
+		std::vector<announce_endpoint> endpoints;
```

Comment:
I'm still weighing this. On the upside flattening the list makes it easier to maintain API compatibility in the python bindings, but it also makes the public API more awkward since clients are likely to want to group trackers by url for displaying to the user. It creates some weird corner cases, like if for some reason all listen sockets went away momentarily we might lose the tracker information. It would also result in the duplication of tracker specific fields like url, trackerid, teir, and source which annoys the DBA in me. If I don't flatten the list I think I'll at least move all of the dynamic state to `announce_endpoint` and treat them as separate trackers for the purpose of announcing.